### PR TITLE
broker: force refresh token at login when online

### PR DESCRIPTION
**Pull request by shiv-tyagi**
Closes ubuntu/authd#614
Setting token expiry to one hour in the past to make sure it is always refreshed at login when online.
Thanks @adombeck for suggesting how to do it. Please review.

### CODE REVIEW

File: internal/broker/broker.go [hunk 1]
```diff
@@ -557,9 +557,8 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
...
 		}
 
-		// Refresh the token if it has expired and we're online. Ideally, we should always refresh it when we're online,
-		// but the oauth2.TokenSource implementation only refreshes the token when it's expired.
-		if !authInfo.Token.Valid() && !session.isOffline {
+		// Refresh the token if we're online even if the token has not expired
+		if !session.isOffline {
 			authInfo, err = b.refreshToken(ctx, session.authCfg.oauth, authInfo)
 			if err != nil {
 				slog.Error(err.Error())
```
[the remark this answers is not in the document]
Comment (denisonbarbosa) [reply internal/broker/broker.go:564]: As we discussed in the meeting, I think it's best we keep the current behavior. It does show that we need to fine tune the whole "offline authentication" experience, though, so let's work on that!

**Review by denisonbarbosa — APPROVED** [review]

**Comment by shiv-tyagi** [comment]
Thanks for approving @adombeck @denisonbarbosa.

This PR was failing few QA tests earlier. I saw few tests were fixed in main branch. I've rebased and force pushed my branch.

**Comment by shiv-tyagi** [comment]
@adombeck @denisonbarbosa Sorry for bugging you. Could you please look into the tests which are failing and help me figure out if I did break anything?

**Comment by adombeck** [comment]
> Could you please look into the tests which are failing and help me figure out if I did break anything?

So the test that fails is `TestIsAuthenticated/Successfully_authenticate_user_with_password` and it's failing because the test expects the token that's stored in the data directory to contain the user ID `saved-user-id` but it contains the user ID `test-user-id` instead. That's because with this change, the token is refreshed in this test but previously it was not. The refreshed token comes from [this mocked identity provider](https://github.com/shiv-tyagi/authd-oidc-brokers/blob/2dac37475b8723600ed758c6227576a9abb9587f/internal/testutils/provider.go#L179-L179), which uses the user ID `test-user-id`.

The fix is to update the golden file of that test. I'll push a commit which does that.

**Comment by adombeck** [comment]
The `TestConcurrentIsAuthenticated` test is also failing, and the fix for that is more complex. I'll take over from here and let @denisonbarbosa review my commits.

**Comment by shiv-tyagi** [comment]
> I'll take over from here and let @denisonbarbosa review my commits.

Cool. Thanks again. I'll go hunting for more issues :)

**Review by denisonbarbosa — APPROVED** [review]

**Pull request merged**